monai/apps/auto3dseg/hpo_gen.py

```diff
@@ -227,3 +229,164 @@ def run_algo(self, obj_filename: str, output_folder: str = ".", template_path=No
+            and supports saving to and loading from pickle files via ``algo_from_pickle`` and ``algo_to_pickle``.
+        params: a set of parameter to override the algo if override is supported by Algo subclass.
+
+    Examples:
+        The experiment will keep generating new folders to save the model checkpoints, scripts, and configs if available.
+        ├── algorithm_templates
+        │   └── unet
+        ├── unet_0
+        │   ├── algo_object.pkl
+        │   ├── configs
+        │   └── scripts
+        ├── unet_0_learning_rate_0.01
+        │   ├── algo_object.pkl
+        │   ├── configs
+        │   ├── model_fold0
+        │   └── scripts
+        └── unet_0_learning_rate_0.1
+            ├── algo_object.pkl
+            ├── configs
+            ├── model_fold0
+            └── scripts
+
+    Notes:
+        Different from NNI and NNIGen, OptunaGen and Optuna can be ran within the Python process.
+
+    """
+
+    def __init__(self, algo=None, params=None):
+        self.algo: Algo
+        self.obj_filename = ""
+
+        if algo is not None:
+            if isinstance(algo, BundleAlgo):
+                if params is None:
+                    self.algo = algo
+                else:
+                    self.algo = deepcopy(algo)
+                    name = os.path.basename(algo.get_output_path()) + "_override"
+                    output_folder = os.path.dirname(algo.get_output_path())
+
+                    params.update({"fill_with_datastats": False})  # just copy, not using datastats to fill
+                    self.algo.export_to_disk(output_folder, name, **params)
+            else:
+                self.algo = algo
+
+            if isinstance(algo, BundleAlgo):
+                self.obj_filename = algo_to_pickle(self.algo, template_path=self.algo.template_path)
+            else:
+                self.obj_filename = algo_to_pickle(self.algo)
+                # nni instruction unknown
+
+    def get_obj_filename(self):
+        """Return the dumped pickle object of algo."""
+        return self.obj_filename
+
+    def get_hyperparameters(self):
+        """
+        Get parameter for next round of training from optuna trial object.
+        This function requires user rewrite during usage for different search space.
+        """
+        if has_optuna:
+            logger.info("Please rewrite this code by creating a child class")
+            return {"learning_rate": self.trial.suggest_float("learning_rate", 0.0001, 0.1)}
+        else:
+            warn("Optuna is not detected. The code will continue to run without Optuna.")
+            return {}
+
+    def set_score(self, acc):
+        self.acc = acc
+
+    def set_trial(self, trial):
+        self.trial = trial
+
+    def print_bundle_algo_instruction(self):
+        pass
+
+    def __call__(self, trial, obj_filename: str, output_folder: str = ".", template_path=None):
```

Comment:
Could you please help add the doc-string here?

Thanks.

Reply:
Added comments. Thanks